modules/transports/core/vfs/src/main/java/org/apache/synapse/transport/vfs/PollTableEntry.java

```diff
@@ -197,43 +236,59 @@ public Integer getFileProcessingCount() {
 		return fileProcessingCount;
 	}
 
-	private void setMoveAfterProcess(String moveAfterProcess) {
+	private void setMoveAfterProcess(String moveAfterProcess) throws AxisFault {
         if (moveAfterProcess == null) {
             this.moveAfterProcess = null;
         } else if (moveAfterProcess.startsWith(VFSConstants.VFS_PREFIX)) {
             // to recover a good directory location if user entered with the vfs: prefix
             // because transport uris are given like that
-            this.moveAfterProcess = moveAfterProcess.substring(VFSConstants.VFS_PREFIX.length());
+            this.moveAfterProcess = resolveHostAtDeployment(moveAfterProcess.substring(VFSConstants.VFS_PREFIX.length()));
         } else {
-            this.moveAfterProcess = moveAfterProcess;
+            this.moveAfterProcess = resolveHostAtDeployment(moveAfterProcess);
         }
     }
 
     public String getMoveAfterErrors() {
+        if (resolveHostsDynamically) {
+            try {
+                return VFSUtils.resolveUriHost(moveAfterErrors);
+            } catch (UnknownHostException | FileSystemException e) {
+                log.warn("Unable to resolving hostname of transport.vfs.MoveAfterErrors: " +
```

Comment:
According to pattern used in this module, warning is used in most similar cases. Here warn is used due to this exception doesn't halt the continiuation of the functionality and just to find root causse if failed later.